mdbook/src/15-interrupts/README.md

```diff
@@ -0,0 +1,83 @@
+## Interrupts
+
+So far, we've gone though a fair bunch of topics about embedded software.  We've read out buttons,
+waited for timers, done serial communication, and talked to other things on the Microbit board using
+I2C.  Each of these things involved waiting for one or more peripherals to become ready. So far, our
+waiting was by "polling": repeatedly asking the peripheral if it's done yet, until it is.
+
+Seeing as our microcontroller only has a single CPU core, it cannot do anything else while it
+waits. On top of that, a CPU core continuously polling a peripheral wastes power, and in a lot of
+applications, we can't have that. Can we do better?
+
+Luckily, we can. While our little microcontroller can't compute things in parallel, it can easily
+switch between different tasks during execution, responding to events from the outside world. This
+switching is done using a feature called "interrupts"!
+
+Interrupts are aptly named: they allow peripherals to actually interrupt the core program execution
+at any point in time. On our MB2's nRF52833, peripherals are connected to the core's Nested Vectored
+Interrupt Controller (NVIC). The NVIC can stop the CPU in its tracks, instruct it to go do something
+else, and once that's done, get the CPU back to what it was doing before it was interrupted. We'll
+cover the Nested and Vectored parts of the interrupt controller later: let's first focus on how the
+core switches tasks.
+
+### Handling Interrupts
+
+Computation is always contextual: the core always needs memory to load inputs and store outputs to.
+Our microcontroller is of what's known as a load-store-architecture, and as such the core does not
```

Comment:
The explanation of a load-store-architecture is a bit weird. Instead of explaining what it doesn't do confusingly using the words store and load, I'd turn it around and say: it's a load and store architecture because typically the data needs to be loaded from memory into registers before doing computation and store the results back to memory afterwards. I also wouldn't refer to registers as "scratch pad memory" but rather something like in-core storage locations.